src/browser/cdp/selectivity/utils.ts

```diff
@@ -165,20 +165,21 @@ const warnUnsupportedProtocol = memoize((protocol: string, dependency: string):
 });
 
 /**
- * @param cssDependencies SORTED uniq array of css dependenciy URI's
- * @param jsDependencies SORTED uniq array of js dependenciy URI's
+ * @param cssDependencies set of css dependenciy URI's
+ * @param jsDependencies set of js dependenciy URI's
  * @returns sorted uniq arrays of relative paths
  */
 export const transformSourceDependencies = (
-    cssDependencies: string[],
-    jsDependencies: string[],
+    cssDependencies: Set<string> | null,
+    jsDependencies: Set<string> | null,
+    mapDependencyPathFn?: null | ((relativePath: string) => string | void),
 ): NormalizedDependencies => {
     const nodeModulesLabel = "node_modules/";
-    const css: string[] = [];
-    const js: string[] = [];
-    const modules: string[] = [];
+    const cssSet: Set<string> = new Set();
+    const jsSet: Set<string> = new Set();
+    const modulesSet: Set<string> = new Set();
```

Comment:
We still need those additional sets since multiple files can be mapped to same files (or modules, which is more likely